modular_skyrat/modules/customization/modules/surgery/organs/ipc.dm

```diff
@@ -170,44 +170,61 @@
 	if(!istype(target, /obj/machinery/power/apc) || !ishuman(user) || !proximity_flag)
 		return ..()
 	user.changeNext_move(CLICK_CD_MELEE)
-	var/obj/machinery/power/apc/A = target
-	var/mob/living/carbon/human/H = user
-	var/obj/item/organ/stomach/robot_ipc/cell = locate(/obj/item/organ/stomach/robot_ipc) in H.internal_organs
+	var/obj/machinery/power/apc/target_apc = target
+	var/mob/living/carbon/human/ipc = user
+	var/obj/item/organ/stomach/robot_ipc/cell = locate(/obj/item/organ/stomach/robot_ipc) in ipc.internal_organs
 	if(!cell)
-		to_chat(H, "<span class='warning'>You try to siphon energy from the [A], but your power cell is gone!</span>")
+		to_chat(ipc, "<span class='warning'>You try to siphon energy from the [target_apc], but your power cell is gone!</span>")
 		return
 
-	if(A.cell && A.cell.charge > 0)
-		if(H.nutrition >= NUTRITION_LEVEL_WELL_FED)
+	if(target_apc.cell && target_apc.cell.charge > 0)
+		if(ipc.nutrition >= NUTRITION_LEVEL_WELL_FED)
 			to_chat(user, "<span class='warning'>You are already fully charged!</span>")
 			return
 		else
-			powerdraw_loop(A, H)
+			powerdraw_loop(target_apc, ipc)
 			return
 
 	to_chat(user, "<span class='warning'>There is no charge to draw from that APC.</span>")
 
-/obj/item/apc_powercord/proc/powerdraw_loop(obj/machinery/power/apc/A, mob/living/carbon/human/H)
-	H.visible_message("<span class='notice'>[H] inserts a power connector into the [A].</span>", "<span class='notice'>You begin to draw power from the [A].</span>")
-	while(do_after(H, 10, target = A))
-		if(loc != H)
-			to_chat(H, "<span class='warning'>You must keep your connector out while charging!</span>")
+#define IPC_CHARGE_MAX 150
+#define IPC_CHARGE_MIN 50
+#define IPC_CHARGE_PER_NUTRITION 10
+#define IPC_CHARGE_DELAY_PER_100 10
+
+/obj/item/apc_powercord/proc/powerdraw_loop(obj/machinery/power/apc/target_apc, mob/living/carbon/human/user)
+	user.visible_message("<span class='notice'>[user] inserts a power connector into the [target_apc].</span>", "<span class='notice'>You begin to draw power from the [target_apc].</span>")
+
+	while(TRUE)
+		var/power_needed = NUTRITION_LEVEL_WELL_FED - user.nutrition // How much charge do we need in total?
+		// Do we even need anything?
+		if(power_needed <= 0)
+			to_chat(user, "<span class='notice'>You are fully charged.</span>")
 			break
-		if(A.cell.charge == 0)
-			to_chat(H, "<span class='warning'>The [A] doesn't have enough charge to spare.</span>")
+		// Is the APC almost empty?
+		if(target_apc.cell.percent() < 10)
+			to_chat(user, "<span class='warning'>[target_apc]'s emergency power is active.</span>")
 			break
-		A.charging = 1
-		if(A.cell.charge >= 500)
-			do_sparks(1, FALSE, A)
```

Comment:
Is this now removed? Seems like a somewhat notable balance change if it is

Reply:
All that code did is show sparks each charge cycle, I can add it back if you want

Reply:
I think sparks when charging is good.

Reply:
Add it back yes.

Reply:
Does do_sparks() not cause fires? I always assumed it did. I don't generally play synths though.

Reply:
It's added back.